hack/vmimage.sh

```diff
@@ -39,11 +34,7 @@ make create
 
 make e2e
 
-# Run the image validation and capture the artifacts
-go generate ./... && go run -ldflags "-X main.gitCommit=$GITCOMMIT" ./cmd/vmimage -buildResourceGroup "${BUILD_RESOURCE_GROUP}" -imageResourceGroup "${IMAGE_RESOURCEGROUP}" -image "${IMAGE_RESOURCENAME}" -imageStorageAccount "${IMAGE_STORAGEACCOUNT}" -validate -preserveBuildResourceGroup
-
-# Currently there are only 3 logs we want to capture
-mv /tmp/{yum_update_info,yum_check_update,scap-report.html} ${ARTIFACTS:-/tmp}/ || true
+make vmimage-validate
```

Comment:
Will see if that works. I currently see no reason for failure 🤞

Reply:
This would make life easier and simplified if it does but I think the exit of the vmimage-validate.sh might trigger the cleanup.  We might want to rename or remove the `trap` here https://github.com/openshift/openshift-azure/blob/master/hack/vmimage-validate.sh#L9 as I think that exit will trigger https://github.com/openshift/openshift-azure/blob/master/hack/vmimage.sh#L3.  Would namespacing the cleanup functions be worth it?  Rename them?

Reply:
> but I think the exit of the vmimage-validate.sh might trigger the cleanup

> Would namespacing the cleanup functions be worth it? Rename them?

Do I understand correctly that your concern is that `trap cleanup EXIT` from `vmimage-validate.sh` will trigger `cleanup` from `vmimage.sh`? If so, then I think it's not the problem because we are not sourcing/including `vmimage-validate.sh` into `vmimage.sh`: we run it as a separate command (`make vmimage-validate`). The process will inherit env vars, but there will be no conflict between function names.

So exit of `vmimage-validate.sh` will trigger cleanup from `vmimage-validate.sh` which just removes `BUILD_RESOURCE_GROUP`:

https://github.com/openshift/openshift-azure/blob/7c5a962d423a9f8e387aea60f37d44af9101adb9/hack/vmimage-validate.sh#L3-L7

And on exit from `vmimage.sh` another cleanup function will be called:

https://github.com/openshift/openshift-azure/blob/7c5a962d423a9f8e387aea60f37d44af9101adb9/hack/vmimage.sh#L3-L12

Reply:
Ok, sgtm.